File: week2/JavaScript/Task01.js
function findAndPrint(messages) {
    // 根據自我介紹中出現的語句判斷，
    // 依照18 years old、college student、legal age、vote for Donald Trump等語句推理，
    // 可以得知Bob、Copper、Leslie、Vivian年紀已超過17歲。因此根據相關關鍵字作為判斷依據。
    // your code here, based on your own rules
    let obj = Object.keys(messages);
    for (let g = 0; g < obj.length; g++) {
        let key = obj[g];
        let value = messages[key];
        let arr = value.split(' ');           //將值轉換為分割字符串
        for (let i = 0; i < arr.length; i++) {      //用if for迴圈判斷字符始有有相符，沒有的話就跳下一個。
            if ("18" == arr[i]) {
                console.log(key);
            } else if ("college" == arr[i]) {
                console.log(key);
            } else if ("legal" == arr[i]) {
                console.log(key);
            } else if ("vote" == arr[i]) {
                console.log(key);
            } else {
                continue;
            };
        };
    }
};

findAndPrint({
    "Bob": "My name is Bob. I'm 18 years old.",
    "Mary": "Hello, glad to meet you.",
    "Copper": "I'm a college student. Nice to meet you.",
    "Leslie": "I am of legal age in Taiwan.",
    "Vivian": "I will vote for Donald Trump next week",
    "Jenny": "Good morning."
});

File: week2/Python/Task01.py
def find_and_print(messages):
# 當文字串中有[18,college,legal,vote]這些字詞的時候print當前的Key. 沒有的話就跳過進到下一輪。
    for i in messages:
        if ("18" in messages[i]):
            print(i)
        elif("college" in messages[i]):
            print(i)
        elif("legal" in messages[i]):
            print(i)
        elif("vote" in messages[i]):
            print(i)
        else:
            continue


find_and_print({
"Bob":"My name is Bob. I'm 18 years old.",
"Mary":"Hello, glad to meet you.",
"Copper":"I'm a college student. Nice to meet you.",
"Leslie":"I am of legal age in Taiwan.",
"Vivian":"I will vote for Donald Trump next week",
"Jenny":"Good morning."
})